src/batterystatus.cpp

```diff
@@ -113,6 +167,37 @@ void BatteryStatusPrivate::registerSignals()
             this, SLOT(chargePercentageChanged(int)));
 }
 
+BatteryStatus::ChargingMode BatteryStatusPrivate::parseChargingMode(int mode)
+{
+    switch (mode) {
+    case MceChargingModeEnable:
+        return BatteryStatus::EnableCharging;
+    case MceChargingModeDisable:
+        return BatteryStatus::DisableCharging;
+    case MceChargingModeApplyThresholds:
+        return BatteryStatus::ApplyChargingThresholds;
+    case MceChargingModeApplyThresholdsAfterFull:
+        return BatteryStatus::ApplyChargingThresholdsAfterFull;
+    }
+
+    return BatteryStatus::EnableCharging;
+}
+
+int BatteryStatusPrivate::chargingModeToInt(BatteryStatus::ChargingMode mode)
+{
+    switch (mode) {
+    default:
+    case BatteryStatus::EnableCharging:
+        return MceChargingModeEnable;
```

Comment:
consistency nit: return from default: here vs. return from outside switch in the above function. Both ways are just fine, but now there are two very similar methods that look different. Feel free to ignore ;-)